performance/profiles/profile_radial_integral.cc

```diff
@@ -0,0 +1,313 @@
+          OptimizationType::Interpolator>>(hypers);
+      this->radial_integral = rc_shared;
+      break;
+    }
+    default:
+      throw std::logic_error(
+          "The desired combination of parameters can not be handled.");
+      break;
+    }
+  }
+
+  template <class StructureManager>
+  void compute(StructureManager & manager) {
+    // specialize based on the type of radial contribution
+    using internal::AtomicSmearingType;
+    using internal::OptimizationType;
+    using internal::RadialBasisType;
+
+    switch (internal::combine_to_radial_contribution_type(
+        this->radial_integral_type, this->atomic_smearing_type,
+        this->optimization_type)) {
+    case internal::combine_to_radial_contribution_type(
+        RadialBasisType::GTO, AtomicSmearingType::Constant,
+        OptimizationType::None): {
+      this->compute_impl<RadialBasisType::GTO, AtomicSmearingType::Constant,
+                         OptimizationType::None>(manager);
+      break;
+    }
+    case internal::combine_to_radial_contribution_type(
+        RadialBasisType::GTO, AtomicSmearingType::Constant,
+        OptimizationType::Interpolator): {
+      this->compute_impl<RadialBasisType::GTO, AtomicSmearingType::Constant,
+                         OptimizationType::Interpolator>(manager);
+      break;
+    }
+    case internal::combine_to_radial_contribution_type(
+        RadialBasisType::DVR, AtomicSmearingType::Constant,
+        OptimizationType::None): {
+      this->compute_impl<RadialBasisType::DVR, AtomicSmearingType::Constant,
+                         OptimizationType::None>(manager);
+      break;
+    }
+    case internal::combine_to_radial_contribution_type(
+        RadialBasisType::DVR, AtomicSmearingType::Constant,
+        OptimizationType::Interpolator): {
+      this->compute_impl<RadialBasisType::DVR, AtomicSmearingType::Constant,
+                         OptimizationType::Interpolator>(manager);
+      break;
+    }
+    }
+  }
+
+  template <internal::RadialBasisType RadialType,
+            internal::AtomicSmearingType SmearingType,
+            internal::OptimizationType OptType, class StructureManager>
+  void compute_impl(std::shared_ptr<StructureManager> manager) {
+    using math::PI;
+    auto radial_integral{
+        downcast_radial_integral_handler<RadialType, SmearingType, OptType>(
+            this->radial_integral)};
+    auto coefs_n = math::Matrix_t(this->max_radial, this->max_angular + 1);
+    auto coefs_c = math::Vector_t(this->max_radial);
+
+    for (auto center : manager) {
+      coefs_c = radial_integral->template compute_center_contribution(center) /
+                std::sqrt(4.0 * PI);
+      for (auto neigh : center.pairs()) {
+        const double & dist{manager->get_distance(neigh)};
+        coefs_n = radial_integral->template compute_neighbour_contribution(
+            dist, neigh);
+      }
+    }
+  }
+};
+
+int main(int argc, char * argv[]) {
```

Comment:
I don't see the point including this file (and corresponding JSON input) in the repository, who would use it? 

It could be nice to have this transformed into a benchmark though, so that we can make sure there are no performance regression. This would require some extra work though, so I'm fine to leave it out.

Reply:
That's how I found out about this bug.

Reply:
That's a quite low probability of being useful again for 350+ new lines of code that we would have to make sure keep compiling as we change the rest of the code